### DISCUSSION TITLE
Compiling libmesh with vtk

### DISCUSSION THREAD

**Discussion by drebbel1z** [idaholab/moose#22975] [Q&A Getting Started] [answered]
I am trying to install libmesh with vtk. 

I have been able to install vtk from source,  but when I input "./scripts/update_and_rebuild_libmesh.sh --with-vtk-lib=~/projects/vtk_install/lib64 --with-vtk-include=~/projects/vtk_install/include/vtk-9.2"  into the terminal, the installation still proceeds without vtk (See below)
```
checking ~/projects/vtk_install/include/vtk-9.2/vtkConfigure.h usability... no
checking ~/projects/vtk_install/include/vtk-9.2/vtkConfigure.h presence... no
checking for ~/projects/vtk_install/include/vtk-9.2/vtkConfigure.h... no
VTK header files not found!
<<< Configuring library without VTK support >>>
```
I checked the named folder and there is no vtkConfigure.h present. I however found vtkPythonConfigure.h. Are these related?

How do I fix this and be able to compile libmesh with vtk?

I already found this site: https://groups.google.com/g/moose-users/c/Kb6s_LWcEnU
It helped me with installing vtk, but I am currently stuck.

I configured vtk using 
 cmake  -DCMAKE_INSTALL_PREFIX=~/projects/vtk_install -DCMAKE_BUILD_TYPE=Release -DVTK_Group_Rendering=OFF -DVTK_WRAP_PYTHON=ON -Wno-dev $HOME/projects/vtk
as suggested by the site.

**Comment by drebbel1z** [comment, marked as answer]
@GiudGiud, any help?

**Reply by GiudGiud** [reply]
Hello

@milljm knows these things better than I do
Why did you pass -Wno-dev to the vtk install?
So this is on a cluster right, you are not using conda anywhere? I d probably just download VTK in a conda environment with just that

Guillaume

**Reply by doquang** [reply]
libmesh requires a VTK which MPI supported. Have you compile your VTK with option `-DVTK_USE_MPI:BOOL=ON` ?

**Reply by drebbel1z** [reply]
@doquang  I just did. I still can't find the vtkConfigure.h file in the include directory

@GiudGiud I tried the conda installation but I have not had much luck with that. That is why I installed vtk separately. wrt to the conda installation, it compiles and builds fine, except for this warning: unsupported GNU_PROPERTY_TYPE. And none of the tests pass.

**Reply by milljm** [reply]
I would not use `~/`... use absolute paths to your `vtk/lib` and `vtk/include/vtk-<version>` directories. Some times `~` are treated as literal characters...
